openmc/universe.py

```diff
@@ -402,22 +402,23 @@ def create_xml_subelement(self, xml_element):
         if len(self._name) > 0:
             element.set("name", str(self._name))
 
-        if isinstance(self._fill, openmc.Material):
-            element.set("material", str(self._fill._id))
+        if isinstance(self.fill, basestring):
+            assert self.fill.strip().lower() == 'void'
```

Comment:
The error checking on the value of `fill` should be performed in the setter rather than here when the XML elements are formed. Also, although I have no issue with `assert`, we've been using the `checkvalue` module for error checking in most places in the Python API, and I'd prefer to keep things consistent.